### PR TITLE
provide configuration settings for AtlasMap URL and Port #30

### PR DESCRIPTION
**Pull request by lhein**
Signed-off-by: Lars Heinemann <lhein.smx@gmail.com>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,7 +28,23 @@
...
+			"title": "AtlasMap Settings",
+			"properties": {
+				"atlasmap.url": {
+					"type": "string",
+					"default": "localhost",
+					"description": "Uri of the AtlasMap instance to use."
+				},
+				"atlasmap.port": {
+					"type": "integer",
+					"default": 3000,
```
Comment (apupier) [inline package.json]: I think it is best to use the 8585 port which is the default one used by AtlasMap

**Comment by lhein** [comment]
atlasmapconfig

This PR simply adds 2 config settings to VSCode. They are not yet evaluated or used anywhere.

File: package.json [hunk 2]
```diff
@@ -28,7 +28,23 @@
...
-		]
+		],
+		"configuration": {
+			"type": "object",
+			"title": "AtlasMap Settings",
+			"properties": {
+				"atlasmap.url": {
+					"type": "string",
+					"default": "localhost",
+					"description": "Uri of the AtlasMap instance to use."
```
Comment (apupier) [inline package.json]: Uri or Url? it is written url everywhere elese

File: package.json (hunk 1 above)
Comment (lhein) [reply package.json]: and the default port to launch a local atlasmap is 3000.

File: package.json (hunk 2 above)
Comment (lhein) [reply package.json]: I have no strong opinion what term to use here. Will chgange it to Url then.

**Comment by lhein** [comment]
> as it is not used, I fear that it can lead to confusion to users that would expect to have this url used.

just don't merge it before the other PRs or done too?

> it is not clear also for the port if it will be used for starting the instance or for opening it or both.

I'd say for both local launching and opening. Otherwise it would become too strange.

File: package.json (hunk 1 above)
Comment (apupier) [reply package.json]: it is 8585, see
https://github.com/jboss-fuse/vscode-atlasmap/blob/511f27af89b7d952dbef465a302cff93e4d09618/src/extension.ts#L15 and https://github.com/atlasmap/atlasmap/blame/master/README.md#L37

For SpringBoot application, if nothign configured, the default port is 8080 https://www.baeldung.com/spring-boot-change-port

where port 3000 is coming from?

**Comment by apupier** [comment]
> > as it is not used, I fear that it can lead to confusion to users that would expect to have this url used.
> 
> just don't merge it before the other PRs or done too?

either have a single PR, or change this one to make ti clear in description that the value are not used yet. in following PR, the description will be updated

**Comment by apupier** [comment]
> > it is not clear also for the port if it will be used for starting the instance or for opening it or both.
> 
> I'd say for both local launching and opening. Otherwise it would become too strange.

I agree on that. I think that it needs to be clear in the description of the property.

File: package.json (hunk 1 above)
Comment (lhein) [reply package.json]: my fault. you are right. will change to 8585....3000 is from the live update

**Comment by lhein** [comment]
atlasmapconfig

How about this?

**Comment by apupier** [comment]
maybe a bit agressive with capital letter and 3 exclamation points but at least it is visible.

**Comment by lhein** [comment]
@apupier I am open for better wording ideas.

**Comment by apupier** [comment]
> I am open for better wording ideas.

for instance:
- [Currently not used, will soon provide more flexibility.]
- [Currently not used]
- [Will be used in upcoming version, stay tuned]

**Pull request merged**